lib/libxdp/libxdp.c

```diff
@@ -3223,6 +3223,43 @@ static int xdp_multiprog__unpin(struct xdp_multiprog *mp)
 	return err;
 }
 
+static void xdp_detach_links(uint32_t ifindex) {
+	struct bpf_link_info link_info;
+	u_int32_t link_info_len = sizeof(link_info), id;
+	int err, fd;
+
+	while (true) {
+		err = bpf_link_get_next_id(id, &id);
+		if (err) {
+			if (errno == ENOENT)
+				break;
+			pr_debug("can't get next link: %s (id %d)", strerror(errno), id);
+			break;
+		}
+
+		fd = bpf_link_get_fd_by_id(id);
+		if (fd < 0) {
+			pr_debug("can't get link by id (%u): %s", id, strerror(errno));
+			continue;
```

Comment:
Not sure we should just be continuing here; if we can't get the link fd for one link, we probably can't get it for the next one either, so better to just bail out (with the same `err = -errno` assignment as above).

Reply:
done